frame/contracts/src/exec.rs

```diff
@@ -69,7 +71,34 @@ impl ExecReturnValue {
 	}
 }
 
-pub type ExecResult = Result<ExecReturnValue, DispatchError>;
+/// Call or instantiate both call into other contracts and pass through errors happening
+/// in those to the caller. This enum is for  the caller to distinguish whether the error
+/// happened during the execution of the callee or during the supervisor code execution.
+#[cfg_attr(test, derive(PartialEq, Eq, Debug))]
+pub enum ErrorOrigin {
+	/// The error happened during the setup phase of the contract that is to be called.
```

Comment:
I rephrased this and renamed the variant. I think it is more accurate now.

Reply:
I am more curious about if this can happen only during the setup phase. What is the setup phase anyway?

Reply:
What I meant by that is everything that happens in `ext.call` or `ext.instantiate` before calling `vm.execute`. It is just to differentiate whether the error occurs inside the call to `vm.execute` or outside. But of course it could also be any other error thrown outside of that `vm.execute`. It is just that currently there is no other place where this can happen. But as I said I changed the wording to include those cases.